test: Modern Polars end-to-end tests

skritsotalakis opened the pull request:
Continuing work that started in #2177 which appears to have been inactive for a long period of time. The original work addressed `pivot_test`, and I have picked it up to add more end-to-end tests from modern Polars (#1945 ). 

Co-authored-by @tylerriccio33 (contributor of #2177 )

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [X] ✅ Test
- [ ] 🐳 Other

## Related issues

- Related issue #1945 
- Closes #2177 ?

## Checklist

- [X] Code follows style guide (ruff)
- [ ] Tests added
- [ ] Documented the changes

skritsotalakis commented on tests/modern_polars/unpivot_test.py:
```diff
@@ -0,0 +1,48 @@
+
+
+def test_unpivot(constructor: Constructor, request: pytest.FixtureRequest) -> None:
+    context = (  # a. sqlframe is expected to be case-insensitive?, b. dask dates are in string-format
```
I chose to silence those errors as I am not sure if this is expected behavior or not.
<img width="1860" height="483" alt="Screenshot from 2025-07-22 13-00-22" src="https://github.com/user-attachments/assets/35fbc309-4cc1-47d5-bbc7-df598328e914" />
<img width="1860" height="312" alt="Screenshot from 2025-07-22 13-04-46" src="https://github.com/user-attachments/assets/fb4794f3-382d-4fdd-b8f5-3e7af7922261" />

MarcoGorelli commented:
awesome, thanks @skritsotalakis !

small request:
- can we make the input data (and the expected data) in `unpivot_test` all lower-case?
- can we use `datetime` instead of `date`?

like this we avoid the errors for some backends, whilst staying true to the spirit of the test

skritsotalakis commented:
> awesome, thanks @skritsotalakis !
> 
> small request:
> 
>     * can we make the input data (and the expected data) in `unpivot_test` all lower-case?
> 
>     * can we use `datetime` instead of `date`?
> 
> 
> like this we avoid the errors for some backends, whilst staying true to the spirit of the test

Hey Marco! 

Thanks for the input. I addressed your comments in the latest commit.

skritsotalakis commented:
On another note, is it beneficial to have query tests from modern Polars examples?

Example:
```
    expr= (
    [
        nw.col("time"),
        close.rolling_mean(3).over(order_by='time').alias("3D MA"),
        close.alias("Expanding Average").cum_sum().over(order_by='time') / (close.cum_count().over(order_by='time')+ 1),
        close.ewm_mean(alpha=0.03).over(order_by='time').alias("EWMA($\\alpha=.03$)"),
    ]
    )
    results = df.select(expr)
    ```

MarcoGorelli reviewed (approved):
thanks @skritsotalakis , great work!

yeah the ewm_mean example looks good too!

![](https://i.fluffy.cc/7WXqprWzDtCpbfprBZG1Hngnp7H7pfQm.gif)
